src/python/qgspythonutilsimpl.cpp

```diff
@@ -169,8 +178,7 @@ void QgsPythonUtilsImpl::initPython( QgisInterface* interface )
   // initialize 'iface' object
   runString( "qgis.utils.initInterface(" + QString::number(( unsigned long ) interface ) + ")" );
 
-  QString startuppath = homePythonPath() + " + \"/startup.py\"";
-  runString( "if os.path.exists(" + startuppath + "): from startup import *\n" );
+  runString( QString("qgis.user.startup(%1)").arg( homePythonPath() ) );
```

Comment:
I would put import qgis.user after initInterface() - to be the last thing done within initPython(). The call to qgis.user.startup(homedir) seems redundant as the code can be run when the module is first imported - and the homePythonPath() can be retrieved from QgsApplication...